### PR TITLE
Resize handler detection should not be active when moving multiple elements

### PR DESCRIPTION
**Pull request by enzoferey**
Fixes https://github.com/excalidraw/excalidraw/issues/671.

I also refactored some code related to getting selected elements.

Kapture 2020-02-16 at 9 44 57

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1047,7 +1048,8 @@ export class App extends React.Component<any, AppState> {
                   resizingElement: resizeElement ? resizeElement.element : null,
                 });
 
-                if (resizeElement) {
+                const selectedElements = getSelectedElements(elements);
+                if (selectedElements.length === 1 && resizeElement) {
```
Comment (enzoferey) [inline src/index.tsx]: This was the missing check and what solves the bug.

File: src/index.tsx [hunk 2]
```diff
@@ -1043,11 +1044,15 @@ export class App extends React.Component<any, AppState> {
                   { x, y },
                   this.state.zoom,
                 );
-                this.setState({
-                  resizingElement: resizeElement ? resizeElement.element : null,
-                });
 
-                if (resizeElement) {
+                const selectedElements = getSelectedElements(elements);
+                if (selectedElements.length === 1 && resizeElement) {
```
Comment (enzoferey) [inline src/index.tsx]: This was the missing check and what solves the bug.

**Comment by dwelle** [comment]
Is this still an issue on current production? Can't repro.

**Comment by enzoferey** [comment]
Yes it is. Try grabbing 2 selected elements by one of their corners right at the edge before the cursor comes back to normal.

**Comment by dwelle** [comment]
Ok, now I can repro. Let's fix the conflicts and we can merge.

**Comment by enzoferey** [comment]
Done @dwelle. Replaced 3 more hardcoded `filter` I didn't catch somehow before.

**Comment by dwelle** [comment]
Cool. Thanks :+1:

**Pull request merged**